wordpress-mobile/WordPress-Android
# Integrate About Automattic Library

This PR integrates About Automattic library into WPAndroid and uses its implementation to replace current About Activity when feature is enabled.  Also includes setup for composite local build.


Fixes #15529

<img width=320 src="https://user-images.githubusercontent.com/990349/142388993-3f8f025b-976e-478d-b626-c4e0f636c912.png" />


To test:
- Enable `UnifiedAboutFeatureConfig` from Me -> App Settings -> Debug Settings and restart
- Go to Me -> App Settings -> scroll down and click on WordPress for Android under About the app
- See the new About from the library displayed
- Click on Legal and more, it navigates to Legal and More screen

NOTE:  After much wrangling a number of build issues, lint checks and checkstylle issues have been fixed in this PR and also in a separate [PR](https://github.com/wordpress-mobile/WordPress-Android/pull/15602) that is merged in.

## Regression Notes
1. Potential unintended areas of impact
Fixed all known issues

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
Existing unit tests

PR submission checklist:
